Add distance filter

daekoon opened the pull request:
## Changes
- Changed search bar to use HTML 5's search field
- Added distance filter
- Expanding top card shows the scrollbar, and collapsing it hides the scrollbar

Sample deployed on staging: https://dlim-sps-summer20.appspot.com/

daekoon commented:
I just realised that the HTML5's search field doesnt work on mobile. Let me try to fix this

lolfuljames commented on src/main/webapp/style.css:
```diff
@@ -60,35 +60,20 @@ h1 {
   background-color: #fff;
   pointer-events: all;
-  overflow: hidden;
+  overflow-x: hidden;
+  overflow-y: auto;
```
We should maintain the overflow to hidden. Instead change the max height for the pac card so that it can accommodate your changes. It would be best to avoid scrollbars especially in mobile as it isn't as obvious as it seems on the pc page.

daekoon replied:
My concern is that later on when we add more filters the pac card will start to completely fill up the screen on the mobile phone - shouldn't we be avoiding it?

lolfuljames replied:
I agree, no worries, we can keep it then. I'll work on making the filter into a selectable list so that we don't take up so much space on that

lolfuljames commented on src/main/webapp/index.html:
```diff
@@ -71,7 +70,13 @@ <h1>EatGoWhere</h1>
                 </div>
               </fieldset>
-              <div id="price-filter-container" width="100%"">
+              <fieldset>
+                <legend>Maximum Distance</legend>
+                <div id ="distance-filter-container" width="100%">
+                  <input id="distance-filter" type="number" step="100" min="100" value="500"> meters
+                </div>
+              </fieldset>
+              <div id="price-filter-container" width="100%">
```
Have you experimented using sliders to implement this? It will be more intuitive and user-friendly.

daekoon replied:
My concern was that for sliders I have to set a min and max value, and I couldn't think of a good value to set for those two. If I set the range to be too small, it makes the filter redundant - on the other hand, if I set it to be too big, the sliders become too sensitive.

lolfuljames replied:
What do you think about having the range set to 100m - 1000(or up to 1500)m, with steps of 100?

Any less for the min distance it might make it always generate no results.
Any further for the max distance it'll make more sense for them to just change the origin location instead.

daekoon replied:
Changed distance filter to slider with range 100m-2000m

daekoon commented:
Reverted searchbar change.
To fix the issue with the clear button sticking with the scroll, I moved the scrollbar to the inner div containing the filters. Now that I've changed the distance filter to slider, it doesn't trigger the scroll, but in the future when we add more filters it should work nicely.

lolfuljames commented:
Great! It would be good to make both filters look abit more coherent/consistent, could you play around with the padding and margins?
